pkg/frontend/admin_platformworkloadidentityroleset_list_test.go

```diff
@@ -0,0 +1,159 @@
+										RoleDefinitionID:   "/providers/Microsoft.Authorization/roleDefinitions/0336e1d3-7a87-462b-b6db-342b63f7802c",
+										ServiceAccounts: []string{
+											"openshift-ingress-operator:ingress-operator",
+										},
+									},
+								},
+							},
+						},
+					},
+					&api.PlatformWorkloadIdentityRoleSetDocument{
+						PlatformWorkloadIdentityRoleSet: &api.PlatformWorkloadIdentityRoleSet{
+							Properties: api.PlatformWorkloadIdentityRoleSetProperties{
+								OpenShiftVersion: "4.15",
+								PlatformWorkloadIdentityRoles: []api.PlatformWorkloadIdentityRole{
+									{
+										OperatorName:       "CloudControllerManager",
+										RoleDefinitionName: "Azure RedHat OpenShift Cloud Controller Manager Role",
+										RoleDefinitionID:   "/providers/Microsoft.Authorization/roleDefinitions/a1f96423-95ce-4224-ab27-4e3dc72facd4",
+										ServiceAccounts: []string{
+											"openshift-cloud-controller-manager:cloud-controller-manager",
+										},
+									},
+								},
+							},
+						},
+					},
+				)
+			},
+			wantStatusCode: http.StatusOK,
+			wantResponse: &admin.PlatformWorkloadIdentityRoleSetList{
+				PlatformWorkloadIdentityRoleSets: []*admin.PlatformWorkloadIdentityRoleSet{
+					{
+						Properties: admin.PlatformWorkloadIdentityRoleSetProperties{
+							OpenShiftVersion: "4.14",
+							PlatformWorkloadIdentityRoles: []admin.PlatformWorkloadIdentityRole{
+								{
+									OperatorName:       "CloudControllerManager",
+									RoleDefinitionName: "Azure RedHat OpenShift Cloud Controller Manager Role",
+									RoleDefinitionID:   "/providers/Microsoft.Authorization/roleDefinitions/a1f96423-95ce-4224-ab27-4e3dc72facd4",
+									ServiceAccounts: []string{
+										"openshift-cloud-controller-manager:cloud-controller-manager",
+									},
+								},
+								{
+									OperatorName:       "ClusterIngressOperator",
+									RoleDefinitionName: "Azure RedHat OpenShift Cluster Ingress Operator Role",
+									RoleDefinitionID:   "/providers/Microsoft.Authorization/roleDefinitions/0336e1d3-7a87-462b-b6db-342b63f7802c",
+									ServiceAccounts: []string{
+										"openshift-ingress-operator:ingress-operator",
+									},
+								},
+							},
+						},
+					},
+					{
+						Properties: admin.PlatformWorkloadIdentityRoleSetProperties{
+							OpenShiftVersion: "4.15",
+							PlatformWorkloadIdentityRoles: []admin.PlatformWorkloadIdentityRole{
+								{
+									OperatorName:       "CloudControllerManager",
+									RoleDefinitionName: "Azure RedHat OpenShift Cloud Controller Manager Role",
+									RoleDefinitionID:   "/providers/Microsoft.Authorization/roleDefinitions/a1f96423-95ce-4224-ab27-4e3dc72facd4",
+									ServiceAccounts: []string{
+										"openshift-cloud-controller-manager:cloud-controller-manager",
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
```

Comment:
I see some code paths in the function that are not tested, I'd suggest adding a few more test cases here:

1. cosmosdb errors result in 500 Internal Server Error
2. the returned result is sorted in an expected way, even if cosmos returns things not sorted
3. errors in json parsing result in an error

By adding these tests, we'll codify the requirements of the code and make it clear what our needs are (in plain English sentences with some good test names), while also ensuring those features remain in place over time.

Reply:
I agree. I added the first two cases you recommended, but I had trouble figuring out a good way to do the third one... I think it might require using mocks for the API converters or something like that.

Unless you see a quick way to do it, I'd like to leave it for a future refactor, since I couldn't find that case being tested in any other endpoints (and therefore don't have a similar instance to go off of).